### PR TITLE
BaSP-TG-77: Form employee with react hook form

### PR DESCRIPTION
**Pull request by Maxi-SR**
- Implement Input component
- Implement Joi validations
- Implement React hook forms
- Create a file named validations.jsx inside Form folder and create the schema using joi inside
- Use joiResolver() from React hook forms
- Add absolute routes when is neccesary

### CODE REVIEW

**Review by PilarFernandezC — APPROVED** [review]
Good job Maxi

**Review by mara666 — APPROVED** [review]
Good job Maxi, it's okey for me!

**Review by Juanoauil — APPROVED** [review]

File: src/Components/Employees/Form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  });
+
+  const history = useHistory();
+  const params = useParams();
+  const dispatch = useDispatch();
+  const { message, list: employees, isPending } = useSelector((state) => state.employees);
+  const idEdit = params.id;
+  const [showModal, setShowModal] = useState({ success: false, error: false });
+
+  useEffect(async () => {
```
[the remark this answers is not in the document]
Comment (Maxi-SR) [reply src/Components/Employees/Form/index.jsx]: done

**Comment by cami-diaz** [comment]
Hi Max! The message should be "Employee edited successfully"


image

**Comment by cami-diaz** [comment]
Max! does not pass first and last name validations. You can only enter a name and a surname, that is, if my name is Maria Camila San Martin, the validation does not pass. You should be able to put spaces between first and/or last names.
- The error message should be "can only contain letters and spaces."
- 
image

**Comment by cami-diaz** [comment]
Hi Max! When editing an employee or adding a new one, if I enter an invalid email, I don't get the error message; however, if I want to edit or create with that invalid email, I get the deleted modal that I attached to it, but if I am Editing an employee by entering the invalid email even though the modal says it has been deleted, the data they remain as they were before, that is, they are not erased. You should be able to see that the error is the invalid email and not that the employee was deleted.

image

**Pull request merged**